src/NLog.Extensions.Configuration/ConfigSettingLayoutRenderer.cs

```diff
@@ -126,33 +125,48 @@ private IConfigurationRoot LoadFileConfiguration(string fileNames)
                 }
                 else
                 {
-                    configRoot = BuildConfigurationRoot(fileNames);
+                    configRoot = BuildConfigurationRoot(fileNames, IsDefaultConfiguration(FileName, DefaultFileName));
                     _cachedConfigFiles[fileNames] = new WeakReference<IConfigurationRoot>(configRoot);
                     return configRoot;
                 }
             }
         }
 
-        private static IConfigurationRoot BuildConfigurationRoot(string fileNames)
+        private static bool IsDefaultConfiguration(Layout fileName, string defaultFileName)
+        {
+            var layoutFileName = (fileName as SimpleLayout)?.Text;
+            return string.IsNullOrEmpty(layoutFileName) || ReferenceEquals(layoutFileName, defaultFileName);
+        }
+
+        private static IConfigurationRoot BuildConfigurationRoot(string fileNames, bool includeEnvironmentVariables)
         {
             var configBuilder = new ConfigurationBuilder();
+
             string baseDir = null;
             foreach (var fileName in fileNames.Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries))
             {
-                var fullPath = fileName;
-                if (!System.IO.Path.IsPathRooted(fullPath))
+                if (!System.IO.Path.IsPathRooted(fileName) && baseDir == null)
                 {
-                    if (baseDir == null)
-                        baseDir = new BaseDirLayoutRenderer().Render(LogEventInfo.CreateNullEvent()) ?? string.Empty;
-                    fullPath = System.IO.Path.Combine(baseDir, fileName);
+                    baseDir = AddConfigurationBasePath(configBuilder);
                 }
 
-                AddFileConfiguration(configBuilder, fullPath);
+                AddFileConfiguration(configBuilder, fileName);
             }
 
+            if (includeEnvironmentVariables)
+                configBuilder.AddEnvironmentVariables();
```

Comment:
proposal: 

- create a setting for this 
- remove the IsDefaultConfiguration
- disabled by default?

Reply:
Ok I will remove it then